internal/internal_workflow.go

```diff
@@ -1516,8 +1517,11 @@ func setUpdateHandler(ctx Context, updateName string, handler interface{}, opts
 		return err
 	}
 	getWorkflowEnvOptions(ctx).updateHandlers[updateName] = uh
-	if getWorkflowEnvironment(ctx).HandleUpdates(updateName) {
-		getState(ctx).yield("letting any updates waiting on a handler run")
+	if getWorkflowEnvironment(ctx).TryUse(SDKPriorityUpdateHandling) {
+		getWorkflowEnvironment(ctx).HandleQueuedUpdates(updateName)
+		state := getState(ctx)
+		defer state.unblocked()
```

Comment:
The only bit I don't understand now is why we're adding `state.unblocked()` calls after returning from the `yield` here and above.

Reply:
Because this coroutine is no longer blocked after calling `yield`. The scheduler detects a coroutine is blocked if it calls `yield` twice with no `unblocked` between them.

Reply:
Thanks! (I have not yet found time to study this situation to the extent where I feel comfortable adding my approval to the PR.)